### PR TITLE
Remove ClientRealProxy on wasm and testing_aot.

### PR DESCRIPTION
**Pull request by baulig**
Introduce `DISABLE_REMOTING` and `DISABLE_SECURITY` conditionals for wasm and the aot test profiles and remove `ClientRealProxy` from `System.ServiceModel` on those profiles.

### CODE REVIEW

**Comment by baulig** [comment]
@monojenkins commit csproj

File: mcs/class/System.ServiceModel.Web/Test/System.ServiceModel.Web/WebOperationContextTest.cs [hunk 1]
```diff
@@ -48,7 +48,7 @@ namespace MonoTests.System.ServiceModel.Web
 	public class WebOperationContextTest
 	{
 // MonoTouch/FullAOT does not support dynamic proxy code generation.
-#if !MONOTOUCH && !FULL_AOT_RUNTIME
+#if !MOBILE && !FULL_AOT_RUNTIME
```
Comment (marek-safar) [inline mcs/class/System.ServiceModel.Web/Test/System.ServiceModel.Web/WebOperationContextTest.cs]: This should be different define, we can run this on ANDROID

Comment (baulig) [reply mcs/class/System.ServiceModel.Web/Test/System.ServiceModel.Web/WebOperationContextTest.cs]: Well okay, in this case I'll have to introduce that new `DISABLE_SECURITY` conditional from my "Martin removes thing" PR here.

**Comment by baulig** [comment]
We probably need to do csproj again.

**Comment by baulig** [comment]
@marek-safar I made it use the new `DISABLE_REMOTING` conditional and also introduced `DISABLE_SECURITY` (which I will shortly use in separate PR's).

**Comment by baulig** [comment]
@monojenkins build failed

**Pull request merged**